### PR TITLE
Normalize importance values

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
The documentation of `get_params_importance` states that the returned importance values should sum up to 1.0. However, `ShapleyImportanceEvaluator` currently does not normalize its output to 1.0.

It is more natural (and fits to many people's intuition) that the mean absolute SHAP value be *not* normalized to 1.0. We propose to add an option `normalize` to `get_params_importance` that controls whether the returned importance values should be normalized to 1.0. This option is defaulted to `True`, and the normalized version is displayed in `plot_params_importance`.

## Description of the changes
* Add `normalize` option to `get_params_importance`
* Add tests

## Alternative approach
There are possible alternative approaches to this problem.
* Specify `normalize` option in `ShapleyImportanceEvaluator`. This way, `plot_params_importance` can also display the unnormalized values. However, we may have to change the way `<0.01` is displayed in `plot_params_importance`.
* Force all importance values to be normalized/unnormalized.

### CODE REVIEW

File: optuna/importance/__init__.py [hunk 1]
```diff
@@ -75,6 +77,10 @@ def get_param_importances(
                 Specify this argument if ``study`` is being used for multi-objective
                 optimization. For example, to get the hyperparameter importance of the first
                 objective, use ``target=lambda t: t.values[0]`` for the target parameter.
+        normalize:
+            A boolean option to specify whether the sum of the importance values should be
```
Comment (HideakiImamura) [inline optuna/importance/__init__.py:81]: How about making this argument to "experimental"?

File: optuna/importance/__init__.py [hunk 2]
```diff
@@ -24,16 +25,17 @@ def get_param_importances(
...
     The parameter importances are returned as a dictionary where the keys consist of parameter
     names and their values importances.
-    The importances are represented by floating point numbers that sum to 1.0 over the entire
-    dictionary.
-    The higher the value, the more important.
+    The importances are represented by non-negative floating point numbers, where higher values
+    mean that the parameters are more important.
     The returned dictionary is of type :class:`collections.OrderedDict` and is ordered by
     its values in a descending order.
+    If ``normalize`` is :obj:`True`, the sum of the importance values are normalized to 1.0.
```
Comment (HideakiImamura) [inline optuna/importance/__init__.py:38]: How about the following?
```suggestion
    By default, the sum of the importance values are normalized to 1.0.
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have some early comments. PTAL.

**Comment by HideakiImamura** [comment]
@nzw0301 Could you review this PR?

@contramundum53 Could you merge the master to fix the CI error?

File: tests/importance_tests/test_init.py [hunk 3]
```diff
@@ -103,15 +107,19 @@ def objective(trial: Trial) -> float:
 
         # Sanity check for param importances
         assert all(0 <= x < float("inf") for x in param_importance.values())
+        if normalize:
+            assert len(param_importance) == 0 or np.isclose(sum(param_importance.values()), 1.0)
```
Comment (nzw0301) [inline tests/importance_tests/test_init.py:111]: I think the first condition is always false in this test function.
```suggestion
            assert np.isclose(sum(param_importance.values()), 1.0)
```

File: tests/importance_tests/test_init.py [hunk 4]
```diff
@@ -184,6 +206,8 @@ def objective(trial: Trial) -> float:
...
```
Comment (nzw0301) [inline tests/importance_tests/test_init.py:210]: ditto: the first condition is always false in this test function.
```suggestion
            assert np.isclose(sum(param_importance.values()), 1.0)
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Comment by nzw0301** [comment]
This might be out-of-scope in this PR, but let me leave one comment to improve docs. 

In my understanding, `get_param_importances`'s `normalize` flag doesn't affect MDI and fanova importance evaluators since they internally normalise importance score with sklearn (concretely, in `forest.feature_importances_`). So I'm wondering if we could clarify this behaviour on their pages.

**Comment by contramundum53** [comment]
I think we might make `FanovaImportanceEvaluator` return unnormalized values in the future.

**Comment by contramundum53** [comment]
Fixed. Thanks. @nzw0301

File: optuna/importance/__init__.py (hunk 1 above)
Comment (contramundum53) [reply optuna/importance/__init__.py:83]: Fixed. PTAL. @HideakiImamura

**Review by nzw0301 — APPROVED** [review]
Thanks.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**